### PR TITLE
Add Jest-based snapshot testing and Travis

### PR DESCRIPTION
**Pull request by domenic**
Previously we were just kind of running test/test.js as needed and hoping for the best. With this in place we have an easier way of checking for regressions.

Let's get Travis running on the PR first...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  only:
+    - master
+
+notifications:
+  email:
+    on_success: never
+    on_failure: always
+
+script:
+  - npm test
```
Comment (Sebmaster) [inline .travis.yml]: That should be the default so not needed.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+language: node_js
+node_js:
+  - 6
+  - stable
+
+branches:
+  only:
+    - master
```
Comment (TimothyGu) [inline .travis.yml]: This shouldn't be needed either. It's not the default but I don't think it's necessary.

Comment (Sebmaster) [inline .travis.yml]: It should reduce the double travis-build in PRs on direct branches.

Comment (TimothyGu) [inline .travis.yml]: Ah never mind then.

File: .travis.yml (hunk 1 above)
Comment (domenic) [inline .travis.yml]: Yeah I guess I was copying it from somewhere where we also had `npm run lint`

**Pull request merged**